sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -673,6 +677,244 @@ def test_check_estimator_get_tags_default_keys():
     )
 
 
+def test_check_classifiers_multilabel_output_format():
+    n_samples, test_size, n_outputs = 100, 25, 5
+    _, y = make_multilabel_classification(n_samples=n_samples, n_features=2,
+                                          n_classes=n_outputs, n_labels=3,
+                                          length=50, allow_unlabeled=True,
+                                          random_state=0)
+    y_test = y[-test_size:]
+
+    class BaseMultiLabelClassifierMock(ClassifierMixin, BaseEstimator):
+        def __init__(self, response_output):
+            self.response_output = response_output
+
+        def fit(self, X, y):
+            return self
+
+    class MultiLabelClassifierPredict(BaseMultiLabelClassifierMock):
+        def predict(self, X):
+            return self.response_output
+
+    class MultiLabelClassifierPredictProba(BaseMultiLabelClassifierMock):
+        def predict_proba(self, X):
+            return self.response_output
+
+    class MultiLabelClassifierDecisionFunction(BaseMultiLabelClassifierMock):
+        def decision_function(self, X):
+            return self.response_output
+
+    # 1.method predict
+    # 1.1 inconsistent array type
+    clf = MultiLabelClassifierPredict(response_output=y_test.tolist())
+    err_msg = (
+        r"MultiLabelClassifierPredict.predict is expected to output a "
+        r"NumPy array. Got <class 'list'> instead."
+    )
+    assert_raises_regex(
```

Comment:
should we use `pytest.raises` instead since these are tests and not the estimator checks?

Reply:
I think that we have the same limitation for this file. I was thinking that the `run_tests_without_pytest` in this file would force to not use `pytest` but I have to check more in depth if I am wrong.

Reply:
we can use `raises` from our utils here indeed.